src/app/team/documents/proofs/Edit.jsx

```diff
@@ -0,0 +1,160 @@
+import React, { Component, createRef } from 'react';
+
+import Header from '../../../../components/header';
+import Button from '../../../../components/button';
+import Selector from '../../../../components/selector';
+import Input from '../../../../components/input';
+
+const DEFAULT_PROOFS = [
+  'None',
+  'Photo ID',
+  'Proof of address',
+  'Proof of income',
+  'Birth certificates',
+];
+
+class ProofsRequiredEdit extends Component {
+  constructor(props) {
+    super(props);
+
+    let proofs;
+
+    if (this.props.value) {
+      proofs = this.props.value.map(el => el.document);
+    }
+
+    this.state = {
+      proofs,
+      showForm: false,
+    };
+
+    this.customProofRef = createRef();
+    this.updateValue = this.updateValue.bind(this);
+    this.showForm = this.showForm.bind(this);
+    this.onSubmitProofForm = this.onSubmitProofForm.bind(this);
+    this.onProofClick = this.onProofClick.bind(this);
+  }
+
+  onSubmitProofForm(e) {
+    e.preventDefault();
+
+    const newValue = this.customProofRef.current.element.value;
+
+    this.setState(prevState => ({
+      proofs: [
+        ...prevState.proofs,
+        newValue,
+      ],
+      showForm: false,
+    }));
+  }
+
+  onProofClick(name) {
+    let { proofs } = this.state;
+
+    if (name === 'None') {
+      proofs = ['None'];
+    } else {
+      const proofIndex = proofs.findIndex(el => el === name);
+
+      if (proofIndex > -1) {
+        proofs.splice(proofIndex, 1);
+      } else {
+        proofs.push(name);
+      }
+
+      if (proofs.includes('None')) {
+        proofs = proofs.filter(p => p !== 'None');
+      }
+    }
+
+    this.setState({ proofs });
+  }
+
+  proofForm() {
+    return (
+      <form onSubmit={this.onSubmitProofForm}>
+        <div>
+          <Input
+            id="proofs-required-custom-form"
+            ref={this.customProofRef}
+            placeholder="e.g. Proof of identity, proof of address"
+            fluid
+          />
+        </div>
+
+        <Button
+          primary
+          className="mt-3 mb-3"
+          onClick={this.onSubmitProofForm}
+        >
+          OK
+        </Button>
+      </form>
+    );
+  }
+
+  showForm() {
+    this.setState({
+      showForm: true,
+    }, () => {
+      this.customProofRef.current.element.focus();
+    });
+  }
+
+  updateValue = e => this.props.updateValue(
+    this.state.proofs,
+    this.props.id,
+    this.props.metaDataSection,
+    this.props.fieldName,
+  );
+
+  render() {
+    const { state: { proofs, showForm } } = this;
+
+    const totalProofs = [...new Set([
+      ...DEFAULT_PROOFS,
+      ...proofs,
+    ])];
+
+    return (
+      <div className="w-100 WhoDoesItServe">
```

Comment:
Does this actually need styles from `WhoDoesItServe`?
If so, those styles should probably be renamed, and the relevant CSS should be imported.

Reply:
Copy paste error 😅